### PR TITLE
Feature Store 2.0

### PR DESCRIPTION
**Pull request by guderkar**

### CODE REVIEW

File: src/featurestorebundle/databricks/feature/reader/DatabricksFeatureStoreReader.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        logger: Logger,
+        table_names: TableNames,
+        feature_manager: DeltaFeaturesManager,
+    ):
+        self.__logger = logger
+        self.__table_names = table_names
+        self.__features_manager = feature_manager
+
+    def read_latest(self, entity_name: str, feature_names: List[str] = None):
+        full_table_name = self.__table_names.get_latest_full_table_name(entity_name)
```
Comment (kutny) [inline src/featurestorebundle/databricks/feature/reader/DatabricksFeatureStoreReader.py:20]: feature_names = feature_names or []

Comment (kutny) [reply src/featurestorebundle/databricks/feature/reader/DatabricksFeatureStoreReader.py:20]: Add this to address None value for list

Comment (kutny) [reply src/featurestorebundle/databricks/feature/reader/DatabricksFeatureStoreReader.py:20]: Apply to all readers pls

File: src/featurestorebundle/databricks/feature/writer/DatabricksFeatureStoreDataHandler.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+        except ImportError:
+            raise Exception("Cannot import Databricks Feature Store, you need to use ML cluster with DBR 8.3+")
+
+        dbx_feature_store_client = feature_store.FeatureStoreClient()
+        db_name = full_table_name.split(".")[0]
+
+        self.__spark.sql(f"CREATE DATABASE IF NOT EXISTS {db_name}")
+
+        features_data = self.__replace_decimal_types(config.data)
```
Comment (kutny) [inline src/featurestorebundle/databricks/feature/writer/DatabricksFeatureStoreDataHandler.py:36]: This is not a very safe approach :-(

File: src/featurestorebundle/databricks/feature/writer/DatabricksFeatureStoreDataHandler.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        if not self.__table_exists_in_feature_store(fs_client, full_table_name) and not self.__table_exists_in_hive(full_table_name):
+            self.__logger.info(f"Creating new Databricks Feature Store table {full_table_name}")
+            return False
+
+    def __table_exists_in_feature_store(self, dbx_feature_store_client, full_table_name: str):  # noqa
+        try:
+            dbx_feature_store_client.get_feature_table(full_table_name)
+            return True
+
+        except Exception:  # noqa
```
Comment (kutny) [inline src/featurestorebundle/databricks/feature/writer/DatabricksFeatureStoreDataHandler.py:71]: What about checking some more specific exception message?

**Review by kutny — CHANGES_REQUESTED** [review]

**Review by ds-lukaslangr — CHANGES_REQUESTED** [review]

**Pull request merged**